tutorials/images/Images_Tutorial.ipynb

```diff
@@ -0,0 +1,444 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Training Set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from snorkel.learning import GenerativeModel\n",
+    "from scipy import sparse\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "L_train = sparse.csr_matrix(L.T)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Majority Vote**\n",
+    "\n",
+    "To get a sense of how well our heuristic functions are, we calcuate the accuracy of the training set labels if we took the majority vote label for each data point."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mv_labels = np.sign(np.sum(L.T,1))\n",
+    "print ('Coverage of Majority Vote on Train Set: ', np.sum(np.sign(np.sum(np.abs(L.T),1)) != 0)/float(loader.train_num))\n",
+    "print ('Accuracy of Majority Vote on Train Set: ', np.sum(mv_labels == loader.train_ground)/float(loader.train_num))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Generative Model**\n",
+    "We assume that the heuristic functions are conditionally independent of the true label and train the generative model using the labels assigned by the heuristic functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gen_model = GenerativeModel()\n",
+    "\n",
+    "gen_model.train(L.T, epochs=100, decay=0.95, step_size= 0.01/ L.shape[1], reg_param=1e-6)\n",
+    "train_marginals = gen_model.marginals(L_train)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Probabilistic Label Statistics**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.hist(train_marginals, bins=20)\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "learned_table = gen_model.learned_lf_stats()\n",
+    "complete_stats = pd.DataFrame(np.hstack((learned_table.as_matrix()[:,0:2], stats_table.as_matrix()[:,[1,0]])), \n",
+    "                              index = [i.__name__ for i in L_names], \n",
+    "                              columns=['Learn. Acc.', 'Learn. Cov.', 'Emp. Acc.', 'Emp. Cov.'])\n",
+    "complete_stats"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "labels = 2 * (train_marginals > 0.9) - 1\n",
+    "print ('Coverage of Generative Model on Train Set:', np.sum(train_marginals != 0.5)/float(len(train_marginals)))\n",
+    "print ('Accuracy of Generative Model on Train Set:', np.mean(labels == loader.train_ground))"
+   ]
+  },
```

Comment:
@vincentschen you can include a pointer to the noise-aware CNN from the MRI project (the non-video DenseNet)

Reply:
yes -- @jason-fries do you want to point to one of our papers? if so which one?

Reply:
@vincentschen I'd go with @ajratner 's suggestion and just toss in some really generic CNN model (AlexNet or VGG16 from the MRI project). Those models assume no Snorkel infrastructure, just probabilistic labels so they appealing from that angle.

Reply:
Actually, since this is already outside of the Snorkel data structure, I would:
(1) Say explicitly that these are *probabilistic training labels* that can be used to train any model
(2) Say for example in TF, you'd use a cross-entropy loss (https://www.tensorflow.org/versions/r1.1/api_docs/python/tf/nn/sigmoid_cross_entropy_with_logits) to handle the fact that they are probabilistic